azure-go-sdk-conceptual/azure-sdk-go-authorization.md

```diff
@@ -0,0 +1,144 @@
+
+All authentication functions and types are available in the `github.com/Azure/go-autorest/autorest/azure/auth` package.
+
+> [!NOTE]
+> Unless you have special requirements, avoid client-based authentication. This method of authentication encourages bad practices. In particular, using client-based authentication makes it tempting to hard-code credentials. Writing custom code for authentication may also break under future SDK releases if they change authentication requirements.
+
+## Use environment-based authentication
+
+If you're running your application in a tightly controlled environment such as in a container, environment-based authentication is a natural choice. You configure the shell environment before running your application and the Go SDK reads these environment variables at runtime to authenticate with Azure. 
+
+Environment-based authentication has support for all authentication methods except device tokens, evaluated in the following order: Service principal credentials, certificates, username/password, and Managed Service Identity (MSI). If a required environment variable is unset or the SDK gets a refusal from the authentication service, the next authentication type is tried. If the SDK cannot authenticate from the environment, it returns an error.
+
+The following table details the environment variables that need to be set for each authentication type supported by environment-based authentication.
+
+| Authentication type | Environment variable | Description |
+| ------------------- | -------------------- | ----------- |
+| __Service principal__ | `AZURE_TENANT_ID` | The ID for the Active Directory tenant that the service principal belongs to. |
+| | `AZURE_CLIENT_ID` | The name or ID of the service principal. |
+| | `AZURE_CLIENT_SECRET` | The secret associated with the service principal. |
+| __Certificate__ | `AZURE_TENANT_ID` | The ID for the Active Directory tenant that the certificate is registered with. |
+| | `AZURE_CLIENT_ID` | The application client ID associated with the certificate. |
+| | `AZURE_CERTIFICATE_PATH` | The path to the client certificate file. |
+| | `AZURE_CERTIFICATE_PASSWORD` | The password for the client certificate. |
+| __Username/Password__ | `AZURE_TENANT_ID` | The ID for the Active Directory tenant that the user belongs to. |
+| | `AZURE_CLIENT_ID` | The application client ID. |
+| | `AZURE_USERNAME` | The username to log in with. |
+| | `AZURE_PASSWORD` | The password to log in with. |
+| __MSI__ | | MSI does not require any credentials to be set. The application must be running on an Azure resource configured to use MSI. For details, see [Managed Service Identity (MSI) for Azure resources]. |

```

Comment:
after https://github.com/Azure/go-autorest/pull/263  is merged acctually it will have an optional
parameter for user assigned MSI scenario https://github.com/MicrosoftDocs/azure-docs/blob/master/articles/active-directory/pp/msi-how-to-use-vm-msi-token.md

Depends if you want to leave this out and update later

Reply:
Until this is part of an official release, it should be left out. At that point you'll need to inform me of the change.